File: 404-sum-of-left-leaves/sum-of-left-leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    static int ans1=0; 
    public int sumOfLeftLeaves(TreeNode root) {
        if(root==null ||(root.left==null && root.right==null))
        return 0;
        ans1=0;
        //int sum=0;
        fun(root,false);
        return ans1;
    }
    public static void fun(TreeNode root,boolean flag)
    {
        
        if(root==null)
        return ;

        if(root.left==null && root.right==null && flag==true)
        {
            ans1+=root.val;
        }
      
        fun(root.left,true);
        fun(root.right,false);
   
    }
}

File: 404-sum-of-left-leaves/sum-of-left-leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int fun(TreeNode* root, int &sum,bool leftNode){
        if(root==nullptr) return 0;


        if(root->left==nullptr && root->right==nullptr){
            if(leftNode){
            sum+=root->val;
            return sum;
            }
        }
        fun(root->left,sum,true);
        fun(root->right,sum,false);
        return sum;
    }

public:
    int sumOfLeftLeaves(TreeNode* root) {
        if(root==nullptr) return 0;

        int sum=0;
        return fun(root,sum,false);
    }
};